frontend/src/components/chat-interface.tsx

```diff
@@ -58,23 +67,37 @@ export function ChatInterface() {
     if (status === WsClientProviderStatus.ACTIVE) {
       try {
         OpenHands.getRuntimeId().then(({ runtime_id }) => {
-          // eslint-disable-next-line no-console
+          /* eslint-disable-next-line no-console -- Runtime ID is important for debugging */
           console.log(
             "Runtime ID: %c%s",
             "background: #444; color: #ffeb3b; font-weight: bold; padding: 2px 4px; border-radius: 4px;",
             runtime_id,
           );
         });
       } catch (e) {
+        /* eslint-disable-next-line no-console -- Runtime ID error is important for debugging */
         console.warn("Runtime ID not available in this environment");
       }
     }
   }, [status]);
 
+  const { selectedRepository, files: uploadedFiles } = useSelector(
+    (state: RootState) => state.initalQuery,
+  );
+
   const handleSendMessage = async (content: string, files: File[]) => {
-    posthog.capture("user_message_sent", {
-      current_message_count: messages.length,
-    });
+    if (messages.length === 0) {
+      posthog.capture("first_message_after_context", {
+        entry_point: getEntryPoint(selectedRepository, uploadedFiles.length),
+        message_length: content.length,
+        has_repository: !!selectedRepository,
+        has_files: uploadedFiles.length > 0,
+      });
```

Comment:
If we are already capturing whether the user has selected a repository, zip, or direct query, is it necessary to also capture `posthog.capture("repository_selected");` in the `frontend/src/routes/_oh._index/github-repo-selector.tsx` file too?

Reply:
hmm good point though, but i think collecting these allow us to figure out how many time user selected repo/uploaded dir BUT not sent the first instruction.